src/monitor.js

```diff
@@ -17,7 +17,7 @@ class Monitor {
     this._resourceInterval = null;
   }
 
-  async reportError(err, level='error') {
+  async reportError(err, level='error', tags={}) {
```

Comment:
I honestly don't expect people to set tags very often at all. I like the second option you mentioned more out of the two if we do one at all.